scripts/shtf.py

```diff
@@ -0,0 +1,629 @@
+import json
+import logging
+import os
+import time
+from collections import defaultdict
+from dataclasses import dataclass, field
+from datetime import datetime
+from pathlib import Path
+from typing import Callable, DefaultDict, Optional, TypedDict, cast
+
+import click
+import requests
+import toml
+from eth_typing import BlockNumber
+from typing_extensions import NotRequired
+from web3 import Web3
+from web3.constants import ADDRESS_ZERO
+from web3.middleware import geth_poa_middleware
+from web3.types import BlockData
+
+import beamer.contracts
+import beamer.events
+from beamer.config import _merge_dicts
+from beamer.events import ClaimMade, DepositWithdrawn, RequestCreated, RequestFilled
+from beamer.typing import ChainId
+from beamer.util import load_ERC20_abi, setup_logging
+
+logging.disable()
+
+
+class MessageTypes:
+    REQUEST_EXPIRED = "RequestExpired"
+    DEPOSIT_WITHDRAWN = "DepositWithdrawn"
+    UNCLAIMED_FILL = "UnclaimedFill"
+    CHALLENGE_GAME = "ChallengeGame"
+    CHALLENGE_GAME_CLAIM_BY_SOMEONE_ELSE = "ChallengeGameClaimedBySomeoneElse"
+
+
+class MessageType(TypedDict):
+    text: str
+    message_link: NotRequired[str]
```

Comment:
maybe just Optional[str]

Reply:
I first had optional, but mypy was complaining that I was missing a message_link on some places and that is why I changed it to NotRequired. 

I think optional is - it is allowed to not have a value, but the key must be present in the dict.